boards: doc: hwmv2: Update boards' index.rst to us HWMv2 terminology

ajf58 opened the pull request:
The goal of this PR is to improve the developer, contributor, and maintainer experience.

- Developers who read the documentation will find more consistent usage of board terminology
- Contributors and maintainers can avoid churn in PR reviews by giving contributors more examples of preferred 'house style' in the documentation.

An incremental improvement in the documentation is taken here, using a scripted approach. It is a non-goal of this PR to modify every board's documentation to conform to a single style.

See #79621

cc: @kartben 

From the commit message:

HWMv2 introduced new terminology about board 'targets'. This terminology is not reflected in the documentation of many of the boards. Update the documentation to match.

N.b. this is a scripted bulk update: it is intended to pick the low-hanging fruit where the same style/format has been replicated across many documentation files. In pseudo-code (truncated and line wrapped to keep this commit message brief) the script is:
```
find . -type f -name "index.rst" -exec sed -E -i \
    '0,/(esp32c3_042_oled|96b_aerocore2|...|xenvm) board configuration \
    supports the following/s//``\1`` board target supports the \
    following/' {} +
```
where `(esp32c3_042_oled|...|xenvm)` is the complete list of boards output from running `west boards` with a little post-processing to give a pipe-delimited list of boards.

Known limitations include, but aren't limited to:

- It is not expected that this will catch the docuementation of every board, but it should be sufficient for the common use-case where a board contains (as far as Zephyr knows) a single SoC, no board revision (or the documentation is applicable to the default and/or all board revisions).
- This is case-sensitive match.
- Some docs simply refer to 'The board'.

References:
- https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology

nordicjm commented on boards/arduino/portenta_h7/doc/index.rst:
```diff
@@ -35,7 +35,7 @@ More information about STM32H747XIH6 can be found here:
 ==================
 
-The current Zephyr arduino_portenta_h7 board configuration supports the following hardware features:
+The current Zephyr ``arduino_portenta_h7`` board target supports the following hardware features:
```
and this is exactly as I mentioned, this does not exist, this needs to be `arduino_portenta_h7/stm32h747xx/m7`, if talking about the m7 core, or `arduino_portenta_h7/stm32h747xx/m4` if talking about the m4 core, you cannot just do a replacement with the folder name, that is not going to work for a huge number of cases

ajf58 replied:
> this does not exist, this needs to be arduino_portenta_h7/stm32h747xx/m7, if talking about the m7 core, or arduino_portenta_h7/stm32h747xx/m4 if talking about the m4 core

Thanks. 

@benjaminbjornsson ,  can you provide some input as the orginal author? N.b. there's no maintainers listed for this in `MAINTAINERS.yml`

> you cannot just do a replacement with the folder name, that is not going to work for a huge number of cases

This is using the output of `west boards` (which may or may not be just listing directory names). I don't have any numbers to quantify this, so I can't say what 'huge' means.

I'm willing to take a punt that the majority of in-tree boards have a single SoC, no board revision and/or the documentation. I feel like this commit doesn't make any document worse (if it was ambiguous before, then it still is). Perfect is the enemy of good, etc.

ajf58 committed:
boards: doc: hwmv2: Update boards' index.rst to use HWMv2 terminology

HWMv2 introduced new terminology about board 'targets'. This terminology
is not reflected in the documentation of many of the boards. Update the
documentation to match.

N.b. this is a scripted bulk update: it is intended to pick the
low-hanging fruit where the same style/format has been replicated across
many documentation files. In pseudo-code (truncated and line wrapped to
keep this commit message brief) the script is:
```
find . -type f -name "index.rst" -exec sed -E -i \
    '0,/ (esp32c3_042_oled|96b_aerocore2|...|xenvm) board configuration \
    supports the following/s// ``\1`` board target supports the \
    following/' {} +
```

Generating the list of boards:

`(esp32c3_042_oled|...|xenvm)` is the list of boards output from
`west boards`. At the request of reviewers, some boards were excluded:
```
west boards | grep -Ev "(actinius_icarus|actinius_icarus_bee|\
    actinius_icarus_som|actinius_icarus_som_dk|innblue21|\
    innblue22|m5stack_atoms3_lite|lpcxpresso54114|\
    nucleo_h745zi_q|nucleo_h755zi_q|stm32h745i_disco|stm32h747i_disco)$"
```
followed by some post-processing to give a pipe-delimited list.

Known limitations include, but aren't limited to:

- It is not expected that this will catch the documentation of every
board, but it should be sufficient for the common use-case where a board
contains (as far as Zephyr knows) a single SoC, no board revision (or
the documentation is applicable to the default and/or all board
revisions).
- This is case-sensitive match.
- Some docs simply refer to 'The board'.

References:
- https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology

Signed-off-by: Andrew Featherstone <andrew.featherstone@gmail.com>

erwango commented on subsys/net/lib/sockets/sockets.c:
```diff
@@ -74,6 +74,19 @@ static inline void *get_sock_vtable(int sock,
 	return ctx;
 }
 
+size_t msghdr_non_empty_iov_count(const struct msghdr *msg)
+{
+	size_t non_empty_iov_count = 0;
+
+	for (size_t i = 0; i < msg->msg_iovlen; i++) {
+		if (msg->msg_iov[i].iov_len) {
+			non_empty_iov_count++;
+		}
+	}
+
+	return non_empty_iov_count;
+}
+
```
Not related with the change. Please remove

ajf58 replied:
Well spotted, sorry about that. Now fixed in the current `HEAD` of the source branch.

nordicjm commented on boards/innblue/innblue21/doc/index.rst:
```diff
@@ -29,7 +29,7 @@ The following parts are built into the board:
 ==================
 
-The innblue21 board configuration supports the following
+The ``innblue21`` board target supports the following
```
here

ajf58 replied:
Thanks, I've added this to the opt-out list.

nordicjm commented on boards/m5stack/m5stack_atoms3_lite/doc/index.rst:
```diff
@@ -27,7 +27,7 @@ It features the following integrated components:
 ==================
 
-The Zephyr m5stack_atoms3_lite board configuration supports the following hardware features:
+The Zephyr ``m5stack_atoms3_lite`` board target supports the following hardware features:
```
likely here too

ajf58 replied:
Thanks, I've added this to the opt-out list.

nordicjm commented on boards/nxp/lpcxpresso54114/doc/index.rst:
```diff
@@ -47,7 +47,7 @@ For more information about the LPC54114 SoC and LPCXPRESSO54114 board:
 ==================
 
-The lpcxpresso54114 board configuration supports the following hardware
+The ``lpcxpresso54114`` board target supports the following hardware
```
here

ajf58 replied:
Thanks, I've added this to the opt-out list.

JarmouniA commented on boards/st/nucleo_h745zi_q/doc/index.rst:
```diff
@@ -91,7 +91,7 @@ Nucleo H745ZI-Q provides the following hardware components:
 ==================
 
-The Zephyr nucleo_h745zi_q board configuration supports the following hardware
+The Zephyr ``nucleo_h745zi_q`` board target supports the following hardware
```
There are 2 targets for this board.

ajf58 replied:
Thanks, I've added this to the opt-out list.

JarmouniA commented on boards/st/nucleo_h755zi_q/doc/index.rst:
```diff
@@ -84,7 +84,7 @@ Nucleo H755ZI-Q provides the following hardware components:
 ==================
 
-The Zephyr nucleo_h755zi_q board configuration supports the following hardware
+The Zephyr ``nucleo_h755zi_q`` board target supports the following hardware
```
There are 2 targets for this board.

ajf58 replied:
Thanks, I've added this to the opt-out list.

JarmouniA commented on boards/st/stm32h745i_disco/doc/index.rst:
```diff
@@ -67,7 +67,7 @@ More information about STM32H747XIH6 can be found here:
 ==================
 
-The current Zephyr stm32h745i_disco board configuration supports the following hardware features:
+The current Zephyr ``stm32h745i_disco`` board target supports the following hardware features:
```
There are 2 targets for this board.

ajf58 replied:
Thanks, I've added this to the opt-out list.

JarmouniA commented on boards/st/stm32h747i_disco/doc/index.rst:
```diff
@@ -48,7 +48,7 @@ More information about STM32H747XIH6 can be found here:
 ==================
 
-The current Zephyr stm32h747i_disco board configuration supports the following hardware features:
+The current Zephyr ``stm32h747i_disco`` board target supports the following hardware features:
```
There are 2 targets for this board.

ajf58 replied:
Thanks, I've added this to the opt-out list.

ajf58 commented on boards/actinius/icarus/doc/index.rst:
```diff
@@ -147,7 +147,7 @@ nRF9160 pins connected internally:
 ==================
 
-The actinius_icarus board configuration supports the following
+The ``actinius_icarus`` board target supports the following
```
Thanks, I've added this to the opt-out list.

ajf58 commented on boards/actinius/icarus_bee/doc/index.rst:
```diff
@@ -56,7 +56,7 @@ The detailed information about the on-board hardware can be found at the `Icarus
 ==================
 
-The actinius_icarus_bee board configuration supports the following
+The ``actinius_icarus_bee`` board target supports the following
```
Thanks, I've added this to the opt-out list.

ajf58 commented on boards/actinius/icarus_som_dk/doc/index.rst:
```diff
@@ -158,7 +158,7 @@ nRF9160 pins connected internally:
 ==================
 
-The actinius_icarus_som_dk board configuration supports the following
+The ``actinius_icarus_som_dk`` board target supports the following
```
Thanks, I've added this to the opt-out list.

ajf58 commented on boards/innblue/innblue22/doc/index.rst:
```diff
@@ -29,7 +29,7 @@ The following parts are built into the board:
 ==================
 
-The innblue22 board configuration supports the following
+The ``innblue22`` board target supports the following
```
Thanks, I've added this to the opt-out list.

ajf58 commented on boards/actinius/icarus_som/doc/index.rst:
```diff
@@ -56,7 +56,7 @@ The detailed information about the on-board hardware can be found at the `Icarus
 ==================
 
-The actinius_icarus_som board configuration supports the following
+The ``actinius_icarus_som`` board target supports the following
```
Thanks, I've added this to the opt-out list.

jerome-pouiller reviewed (approved):
Looks good for Silabs boards.

JarmouniA reviewed (commented):
I think it would be better & easier to review if you split the changes into commits by vendor.

nashif commented:
maybe this change should be: 'board configuration' -> 'board' and not 'board target', then it will work I think without having to change the identifiers, becase if I am not mistaken, the docmentation is about the board and not any of the qualifiers and variants..

tejlmand commented on boards/arduino/due/doc/index.rst:
```diff
@@ -32,7 +32,7 @@ Hardware
 ==================
 
-The arduino_due board configuration supports the following hardware features:
+The ``arduino_due`` board target supports the following hardware features:
```
this is not a board target, it's the board name.

And the same goes for all other changes in this PR.

The cleanup is appreciated, but we should not swap one bad description for another.
I made a comment in the issue here: https://github.com/zephyrproject-rtos/zephyr/issues/79621#issuecomment-2490674540

Feel free to decide whether to use:
- Full name, in this case: `Arduino Due`
- Board name: in this case: `arduino_due`
- Board target, in this case: `arduino_due/sam3x8e`


I think in this context using the full name, `Arduino Due` looks nicest, but overall using the board name makes most sense because board name is defined for all boards. (Not all boards defines a `Full name` in the board.yml).
And I believe the text in general tries to refer to the high level board name, and not a specific board target.

If you decide to choose board target, then please ensure that a valid / correct board target is used in the text.

tejlmand reviewed (changes requested):
Thanks for the cleanup, it's appreciated, but if doing a cleanup then it should not swap one wrong terminology for another wrong one.

The PR description looks good, as it well describes how you did the changes, but I suggest you remove it from the commit message as it's not really helpful for someone later looking at `git log`. (This suggestion is not blocking for review approval)
```
N.b. this is a scripted bulk update: it is intended to pick the
low-hanging fruit where the same style/format has been replicated across
many documentation files
....
west boards | grep -Ev "(actinius_icarus|actinius_icarus_bee|\
    actinius_icarus_som|actinius_icarus_som_dk|innblue21|\
    innblue22|m5stack_atoms3_lite|lpcxpresso54114|\
    nucleo_h745zi_q|nucleo_h755zi_q|stm32h745i_disco|stm32h747i_disco)$"
...
```

kartben commented:
Hopefully the switch to `.. zephyr:board-supported-hw` (although not all boards have transitioned yet) achieves a similar and more "reliable" result

ajf58 commented:
Let's close this issue as it's overtaken by events.
